Tools/scripts/patchcheck.py

```diff
@@ -64,28 +65,46 @@ def get_git_upstream_remote():
     try:
         subprocess.check_output(cmd,
                                 stderr=subprocess.DEVNULL,
-                                cwd=SRCDIR)
+                                cwd=SRCDIR,
+                                encoding='UTF-8')
     except subprocess.CalledProcessError:
         return "origin"
     return "upstream"
 
 
+def get_git_remote_default_branch(remote_name):
+    """Get the name of the default branch for the given remote
+
+    It is typically called 'main', but may differ
+    """
+    cmd = "git rev-parse --abbrev-ref {}".format(remote_name).split()
+    try:
+        full_name = subprocess.check_output(cmd,
```

Comment:
Hummm, this may return something wrong depending on how you have the git setup. For instance, in my machine this gives:

```
❯ git rev-parse --abbrev-ref origin
origin/HEAD
```

Reply:
Thanks, I have changed the way the branch name is parsed, now it should be (hopefully) more reliable.